Issue #488

Cellzawy opened the pull request:
Attempted to fix #488 

Edited padding for `form-control` class in `bootstrap.css` to prevent the copy and clear buttons from being on top of the translated text. 

Little note: I am a little new to open source so excuse me if I did something wrong.

sushain97 reviewed (changes requested):
Please don't edit bootstrap.css. It's generated from upstream sources. There a number of other CSS files that can be edited instead with specific styles. Or, use inline utility classes.

Cellzawy commented:
> Please don't edit bootstrap.css. It's generated from upstream sources. There a number of other CSS files that can be edited instead with specific styles. Or, use inline utility classes.

Noted, sorry for that. Working on it now

Cellzawy commented:
I have removed my edits on bootstrap.css like you asked, added `pt-4` to both of the `Form.Control`s in `TextTranslationForm.tsx`
This should prevent the buttons from being on top of the text, let me know if you need more spacing though.

ahmedsiam0 commented:
I think more spacing is needed.
When hovering, the button overlaps with the text.
![hover](https://github.com/apertium/apertium-html-tools/assets/97993825/df06dbb9-5731-4567-a0dd-0885c3385656)

Cellzawy commented:
> I think more spacing is needed. When hovering, the button overlaps with the text. ![hover](https://private-user-images.githubusercontent.com/97993825/313505624-df06dbb9-5731-4567-a0dd-0885c3385656.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA3MTM4NTIsIm5iZiI6MTcxMDcxMzU1MiwicGF0aCI6Ii85Nzk5MzgyNS8zMTM1MDU2MjQtZGYwNmRiYjktNTczMS00NTY3LWEwZGQtMDg4NWMzMzg1NjU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE3VDIyMTIzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZkNzJlNzdmNzgwOWY5MmFjOWU0ZjA2ZTViMTk0ZjQxZmNlZjAxMDYzZjNjZDM2M2E1NTQ2NWJkNDgwNjMzMWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ygf1TcjEZthX4GCq_mobsGJTFWhp0GLeydeEeNEce-o)

Fair enough, to increase the spacing I would give it `pr-5` instead of `pr-4`. This is how it would look:
![image](https://github.com/apertium/apertium-html-tools/assets/54197394/8b08e7fd-3296-42a9-8b6b-e79b5a6b98c5)

Or, if `pr-5` it's too much space I can create my own class (let's name it pr-4-5 for example) and give it a padding that's in between `pr-4` and `pr-5`. It would look like this:
![image](https://github.com/apertium/apertium-html-tools/assets/54197394/9c33d841-9a85-484b-80b4-4145b867f667)

Let me know which one you prefer.

ahmedsiam0 commented:
The second option looks visually better for me. (between `pr-4` and `pr-5`).

Cellzawy commented:
> The second option looks visually better for me. (between `pr-4` and `pr-5`).

Done, waiting for your feedback.

sushain97 commented on src/app.css:
```diff
@@ -12,3 +12,7 @@
+
+.pr-4-5 {
+  padding-right: 2.25rem;
+}
```
Let's add this to `translator.css` instead and give it a specific name rather than `pr-4-5`. Perhaps `.translation-text-input`.

sushain97 reviewed (changes requested):
I'm now seeing a large unnecessary padding on the translated text:

![image](https://github.com/apertium/apertium-html-tools/assets/4211838/f9aa6f49-5019-456d-8a78-ccff66a774d7)

Cellzawy commented:
> I'm now seeing a large unnecessary padding on the translated text:
> 
> ![image](https://private-user-images.githubusercontent.com/4211838/313932515-f9aa6f49-5019-456d-8a78-ccff66a774d7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA4NTQ4NDAsIm5iZiI6MTcxMDg1NDU0MCwicGF0aCI6Ii80MjExODM4LzMxMzkzMjUxNS1mOWFhNmY0OS01MDE5LTQ1NmQtOGE3OC1jY2ZmNjZhNzc0ZDcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMTlUMTMyMjIwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWUyMTkyNjMwZmNlMzg1ODQ4MDE3OTYwMTI0OWZkMzM0MzIzZDA1ZGQxNmMwYWIxYzgyNTRmNGJlZmFhZmNlZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.uEeEq4QdFVDdOo0xEJefGiWujXmF8zSCcrS2X3W0-4A)

Right, then instead of using padding and absolute positioning, I am suggesting to make both the textarea and the button components sibling children of a parent div.

This is how it looks:
![image](https://github.com/apertium/apertium-html-tools/assets/54197394/b3273a39-fec2-4037-b270-04b0dea1ee86)
![image](https://github.com/apertium/apertium-html-tools/assets/54197394/00111807-ccce-47df-addf-55748cb4b4bd)

Let me know if you like this approach and I will push my changes.

Cellzawy commented:
> I'm now seeing a large unnecessary padding on the translated text:
> 
> ![image](https://private-user-images.githubusercontent.com/4211838/313932515-f9aa6f49-5019-456d-8a78-ccff66a774d7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA4Njg4MTIsIm5iZiI6MTcxMDg2ODUxMiwicGF0aCI6Ii80MjExODM4LzMxMzkzMjUxNS1mOWFhNmY0OS01MDE5LTQ1NmQtOGE3OC1jY2ZmNjZhNzc0ZDcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMTlUMTcxNTEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDQzZjI1NjNkYjk4YmJkNDcwNzc3OGUzYjU2Y2JiYzQwMjc4NjRhMjg4OWU5ZDIxYTRkMmY4NjQ3ZTEzOWVmMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.P9sQYFmboxyeJ3C0lqSgysrJ645z0-r2y6IMGll4F7A)

Right, apparently that is because of the interface language which is RTL. I have missed that part, sorry.

Fixed it by setting the padding on the left if the interface language is RTL and the opposite if it's LTR:
![image](https://github.com/apertium/apertium-html-tools/assets/54197394/04e5dfba-8113-447e-a414-4549a37626cb)
![image](https://github.com/apertium/apertium-html-tools/assets/54197394/07a6a5b6-a5bf-41b0-a15b-bdba6eeccc51)

Waiting for your feedback.

Cellzawy commented:
> I'm now seeing a large unnecessary padding on the translated text:
> 
> ![image](https://private-user-images.githubusercontent.com/4211838/313932515-f9aa6f49-5019-456d-8a78-ccff66a774d7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA4Njg4MTIsIm5iZiI6MTcxMDg2ODUxMiwicGF0aCI6Ii80MjExODM4LzMxMzkzMjUxNS1mOWFhNmY0OS01MDE5LTQ1NmQtOGE3OC1jY2ZmNjZhNzc0ZDcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMTlUMTcxNTEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDQzZjI1NjNkYjk4YmJkNDcwNzc3OGUzYjU2Y2JiYzQwMjc4NjRhMjg4OWU5ZDIxYTRkMmY4NjQ3ZTEzOWVmMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.P9sQYFmboxyeJ3C0lqSgysrJ645z0-r2y6IMGll4F7A)

Right, apparently it's because of the difference in the interface language's direction, sorry I have missed that part.
Fixed it by setting padding on left if it's an RTL language and the opposite for LTR:
![image](https://github.com/apertium/apertium-html-tools/assets/54197394/c6dc2cbc-6807-406c-9201-adc3f46c50cf)
![image](https://github.com/apertium/apertium-html-tools/assets/54197394/186a4335-529e-4ca7-a3fc-687555f32348)

sushain97 commented on src/components/translator/translator.css:
```diff
@@ -98,3 +98,11 @@ html[dir='rtl'] .translation-modes > :not(:first-child) {
+
+html[dir='rtl'] .translation-text-input {
+  padding-left: 1.85rem;
+}
```
Please add a trailing newline.
